### ISSUE TITLE
ptx: implement -b

### ISSUE THREAD

**Issue by sylvestre** [uutils/coreutils#1763]
GNU man pages says:
```
       -b, --break-file=FILE
              word break characters in this FILE
```

Code is in 
https://github.com/uutils/coreutils/blob/8337179454d0c952746a980ee37a32340f510a46/src/uu/ptx/src/ptx.rs

Tests in
https://github.com/uutils/coreutils/blob/8337179454d0c952746a980ee37a32340f510a46/tests/by-util/test_ptx.rs

**Comment by mike-kfed** [comment]
I gave this a shot, and I _think_ my approach is working:

I read the break-file and create a regex (pseudocode: `format!("[^{}]+", chars_from_breakfile)`) from it.

Problem: output does not match GNU ptx. But I discovered neither does it for other custom regex, the algorithm is somehow broken elsewhere I think. This ptx implementation prepends spaces to `after` [here](https://github.com/uutils/coreutils/blob/cd4003007f447a6cb61a41cb799e30fbc3c092ac/src/uu/ptx/src/ptx.rs#L388) which GNU ptx doesn't seem to do in every case.

```bash
target/debug/ptx -G -W '\w+' tests/fixtures/ptx/input > uu.out
ptx -G -W '\w+' tests/fixtures/ptx/input > gnu.out
diff uu.out gnu.out
```
```diff
1,2c1,2
< .xx "" "and some other like %" "a , b#, c$c" ""
< .xx "" "maybe" "also ~or^" ""
---
> .xx "" "and some other like %" "a, b#, c$c" ""
> .xx "" "maybe" "also~or^" ""
```

you can see the extra space before the comma and before the tilde above. These extra spaces seems to be the only differences in both cases, my break-file and the use of certain regexp passed by `-W`. I'd be motivated to check the algo in general, but I feel it'd be best to try that after PR #1602 was merged otherwise fundamental changes to the algo will get hard to integrate in both trees.

my break-file code, just to show my approach not ready for merge, is here:
https://github.com/uutils/coreutils/compare/master...mike-kfed:ptx_break_file

Please let me know how you prefer to continue from here.

**Comment by allan-silva** [comment]
Is this in progress currently? If not, can I take it ?

**Comment by tertsdiepraam** [comment]
> described problem maybe has went away?

The option is exists, but using it prints:
```
ptx: -b not implemented yet
```

@allan-silva, feel free to pick this up!

**Comment by mike-kfed** [comment]
@tertsdiepraam the described problem has nothing to do with the `-b` option, it is a general problem with whitespace output when using regex. Which then of course also breaks implementing -b using a regex, like I tried in my fork.
But PR 1602 may have fixed that, and then it could be worth going that route.

**Comment by allan-silva** [comment]
Thanks @mike-kfed @tertsdiepraam , I'll take a look.

**Comment by mike-kfed** [comment]
Hi @allan-silva I just found the time to look at my solution again. I merged master into my fork and It seems to work now with the new engine https://github.com/uutils/coreutils/compare/master...mike-kfed:ptx_break_file
my code also already fixes your #3142 bug 
please take a look, if you're happy with it you can take my approach, or parts of it. Or of course I submit my branch as a PR, as you like.

**Comment by mike-kfed** [comment]
> @mike-kfed there is no `master` branch, 

true that, sorry. I renamed the URL to main in my comment.

> but if you finished this work and the tests are passing, it is better to open a PR to people review and comment.

 and will prepare a PR according to the contributing guidelines, thx.
